libs/remote_storage/src/azure_blob.rs

```diff
@@ -312,3 +318,143 @@ impl RemoteStorage for AzureBlobStorage {
         Ok(())
     }
 }
+
+pin_project_lite::pin_project! {
```

Comment:
re: @arpad-m's comment about this being hacky, we should really have a comment here describing: (a) why this type exists, (b) what it does, and (c) how it does it.

Reply:
I wrote a bit like that already before getting to your comment, 978623935 has the latest in next push. I did not include a description of "how", I think the code is quite self-evident and the details are unimportant impl specifics of the sdk, which could change between versions (but luckily we have a test.)

Reply:
[65e3d0c](https://github.com/neondatabase/neon/pull/5446/commits/65e3d0cd3fcc2365dd316b47707cd1513c16d431) is the latest version.